packages/yswaps/scripts/libraries/dexes/solidly.ts

```diff
@@ -75,6 +65,92 @@ export const getBestPathEncoded = async (swapParams: SwapParams): Promise<SwapRe
   return response;
 };
 
+
+async function hopTokenCalculation({ swapParams }: {swapParams: SwapParams}) {
+  const { tokenIn, tokenOut, amountIn, hopTokensToTest } = swapParams;
+  const factory = await ethers.getContractAt<ISolidlyFactory>(ISolidlyFactory__factory.abi, swapParams.solidlyFactory);
+  const router = await ethers.getContractAt<ISolidlyRouter>(ISolidlyRouter__factory.abi, swapParams.solidlyRouter);
+
+  let bestMaxPath: ISolidlyRouter.RouteStruct[] = []
+  let bestAmountOut: BigNumber = BigNumber.from('0');
+  let bestHopToken: string = '';
+  if (!hopTokensToTest) return { bestMaxPath, bestAmountOut };
+
+  for (const hopToken of hopTokensToTest) {
+    let hopMaxPath: ISolidlyRouter.RouteStruct[] = [];
+    let hopAmountOut: BigNumber = BigNumber.from('0');
```

Comment:
There is no need of creating this two I think. You can use `bestMaxPath` and `bestAmountOut`

Reply:
correct, also I just refactored to not duplicate the IF logic to assign amountOut and maxPath